docs/creating-modules-for-deployment.md

```diff
@@ -244,7 +257,8 @@ Because `totalSupply` is not a number, it is a future.
 A deployment can be put `on-hold` until an on-chain event has been emitted (for instance a timelock or multisig approval):
 
 ```tsx
-const multisig = m.deploy("Multisig");
+const artifact = m.getArtifact("Multisig");
+const multisig = m.contract("Multisig", artifact, { args: [] });
```

Comment:
By the same argument this should probably just be `m.contract("Multisig")`.

Reply:
I actually disagree on this one. Multisig is _the_ example where it does matter that we pass in an artifact because `m.event` requires it to be an `ArtifactContract`. I was just lazy on the initial docs for this because it was pseudocode anyways and the full example does show it.

Reply:
Ah and that is because we need to lessen the `m.event` constraint to also include `HardhatContract`s?

Reply:
Originally it needed to be this way because it used the artifact to generate the event param futures. Now that I think about it, though, since we're front-loading the artifacts now, we might not need this constraint anymore

Reply:
Do you want to tackle that here or move that out to a different issue?

Reply:
Created #187 to address this concern